From a06f9cac94bbd549585584b880df810ea7b4eaee Mon Sep 17 00:00:00 2001
From: Prakash Brahmajyosyula <bprakash@marvell.com>
Date: Mon, 28 Jan 2019 14:34:17 +0530
Subject: [PATCH 0907/1051] soc: octeontx2: Add PTP subsystem id for octeontx2
 chip.

This patch adds PCI device subsystem ID of PTP for octoentx2 soc
and also register PTP driver with subsystem id.

Change-Id: I754537df4eadea0f74a512cf2951481c6b263434
Signed-off-by: Prakash Brahmajyosyula <bprakash@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/2829
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/ptp.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2/ptp.c b/drivers/soc/marvell/octeontx2/ptp.c
index 1e7698aee7a3..56bc8aad8302 100644
--- a/drivers/soc/marvell/octeontx2/ptp.c
+++ b/drivers/soc/marvell/octeontx2/ptp.c
@@ -17,6 +17,7 @@
 #define DRV_NAME	"Marvell PTP Driver"
 
 #define PCI_DEVID_OCTEONTX2_PTP		0xA00C
+#define PCI_SUBSYS_DEVID_OCTEONTX2_PTP	0xB200
 #define PCI_DEVID_OCTEONTX2_RST		0xA00E
 
 #define PCI_PTP_BAR_NO	0
@@ -196,7 +197,9 @@ static void ptp_remove(struct pci_dev *pdev)
 }
 
 static const struct pci_device_id ptp_id_table[] = {
-	{ PCI_DEVICE(PCI_VENDOR_ID_CAVIUM, PCI_DEVID_OCTEONTX2_PTP) },
+	{ PCI_DEVICE_SUB(PCI_VENDOR_ID_CAVIUM, PCI_DEVID_OCTEONTX2_PTP,
+			 PCI_VENDOR_ID_CAVIUM,
+			 PCI_SUBSYS_DEVID_OCTEONTX2_PTP) },
 	{ 0, }
 };
 
-- 
2.17.1

